### PR TITLE
fix and run unittests in vscode

### PR DESCRIPTION
**Pull request by chihangwang**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (henry60603) [inline .vscode/settings.json:1]: Thanks for the study and the implementation here.

File: src/utils/llm_endpoint.py [hunk 2]
```diff
@@ -5,44 +5,38 @@
...
-        self.key = key
-        if self.platform == 'openai':
-            if not self.key:
-                self.key = os.getenv('OPENAI_API_KEY')
-                if not self.key:
-                    raise RuntimeError("'OPENAI_API_KEY' not found via os.getenv" )
-            self.client = OpenAI(
-                api_key=self.key
-            )
+        key = None  # key should be set in environment, not in code.
```
Comment (henry60603) [inline src/utils/llm_endpoint.py]: let's remove it since it is not used in the same layer.

Comment (chihangwang) [reply src/utils/llm_endpoint.py]: Take a look below. It is used in the initialization of AI client. My point is just that we shouldn't keep it still accessible after the initialization; it should be only temporary.

Comment (henry60603) [reply src/utils/llm_endpoint.py]: Line 11 is ok to be removed, since it is initialized below for different llm in Line 14 and Line 20

**Review by henry60603 — APPROVED** [review]

**Pull request merged**